### PR TITLE
Consolidate participant creation into 1 place

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Consolidate payment creation into 1 place

Before
----------------------------------------
The credit card part & the non-credit card part of the participant form call participant.create in different places, for historical reasons

After
----------------------------------------
One call. Also this is moved earlier to allow us to create the line items in a saner way / remove the blockers to switching to v3 or v4 order api

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Event/Form/Participant.php [hunk 1]
```diff
@@ -1618,14 +1603,17 @@ protected function addParticipant($contactID) {
...
-      'fee_amount' => $this->order->getTotalAmount(),
       'fee_currency' => $this->getCurrency(),
       'campaign_id' => $this->getSubmittedValue('campaign_id'),
       'note' => $this->getSubmittedValue('note'),
-      'is_test' => ($this->_mode === 'test)'),
+      'is_test' => $this->isTest(),
     ];
+    $order = $this->getOrder();
+    if ($order) {
+      $participantParams['fee_level'] = $this->order->getAmountLevel();
```
Comment (seamuslee001) [inline CRM/Event/Form/Participant.php:1614]: This doesn't look quite right compared to L1615

Comment (eileenmcnaughton) [reply CRM/Event/Form/Participant.php:1614]: Yeah true - it would work fine but it's inconsistent

Comment (eileenmcnaughton) [reply CRM/Event/Form/Participant.php]: I fixed it

**Comment by colemanw** [comment]
She fixed it.

**Pull request merged**